servicetalk-opentracing-http/src/main/java/io/servicetalk/opentracing/http/HttpHeadersKeyValueAccessor.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.opentracing.http;
+
+import io.servicetalk.http.api.HttpHeaders;
+import io.servicetalk.opentracing.inmemory.CharSequenceKeyValueAccessor;
+
+import javax.annotation.Nullable;
+
+import static java.util.Objects.requireNonNull;
+
+final class HttpHeadersKeyValueAccessor implements CharSequenceKeyValueAccessor {
```

Comment:
nit: The typical pattern is to name these `<myType>Adapter` as we see in the open tracing repo: https://github.com/opentracing/opentracing-java/blob/master/opentracing-api/src/main/java/io/opentracing/propagation/TextMapExtractAdapter.java

it may be a good idea to follow that pattern so that users familiar with those apis don't get confused.